plugins/woocommerce-admin/client/inbox-panel/index.js

```diff
@@ -142,44 +175,51 @@ const renderNotes = ( {
 					);
 				} ) }
 			</TransitionGroup>
+			{ allNotesFetched ? null : notesHaveResolved ? (
+				<CardFooter
+					className="wooocommerce-inbox-card__footer"
+					size="medium"
+				>
+					<Button
+						isPrimary={ true }
+						onClick={ () => {
+							loadMoreNotes();
+						} }
+					>
+						{ notesArray.length > DEFAULT_INBOX_QUERY.per_page
+							? __( 'Show more', 'woocommerce' )
+							: __( 'Show older', 'woocommerce' ) }
+					</Button>
+				</CardFooter>
+			) : (
+				<InboxNotePlaceholder className="banner message-is-unread" />
+			) }
 		</Card>
 	);
 };
 
-const INBOX_QUERY = {
-	page: 1,
-	per_page: QUERY_DEFAULTS.pageSize,
-	status: 'unactioned',
-	type: QUERY_DEFAULTS.noteTypes,
-	orderby: 'date',
-	order: 'desc',
-	_fields: [
-		'id',
-		'name',
-		'title',
-		'content',
-		'type',
-		'status',
-		'actions',
-		'date_created',
-		'date_created_gmt',
-		'layout',
-		'image',
-		'is_deleted',
-		'is_read',
-		'locale',
-	],
-};
-
 const InboxPanel = ( { showHeader = true } ) => {
+	const [ noteDisplayQty, setNoteDisplayQty ] = useState(
+		DEFAULT_INBOX_QUERY.per_page
+	);
+	const [ allNotesFetched, setAllNotesFetched ] = useState( false );
+	const [ allNotes, setAllNotes ] = useState( [] );
```

Comment:
This is used to store fetched notes, since we don't want them all to disappear (as happens with the returned `notes` variable with `useSelect`) while fetching more notes. This results in a much better UI experience.